iroh-resolver/src/unixfs.rs

```diff
@@ -492,16 +498,57 @@ impl<T: ContentLoader + Unpin + 'static> AsyncRead for UnixfsContentReader<T> {
     }
 }
 
+impl<T: ContentLoader + Unpin + 'static> AsyncSeek for UnixfsContentReader<T> {
+    fn start_seek(mut self: Pin<&mut Self>, position: std::io::SeekFrom) -> std::io::Result<()> {
+        match &mut *self {
+            UnixfsContentReader::File {
+                root_node: _,
+                pos,
+                pos_max: _,
+                current_node: _,
+                current_links: _,
+                loader: _,
+                out_metrics: _,
+            } => {
+                match position {
+                    std::io::SeekFrom::Start(offset) => {
+                        *pos = offset as usize;
+                    }
+                    std::io::SeekFrom::End(_offset) => {
+                        // This technically means we can seek past the end which we don't support
+                        // TODO: support rolling buffer reads, ie use overflow remainder to read from the start again
+                        todo!()
+                    }
+                    std::io::SeekFrom::Current(offset) => {
+                        *pos = (*pos as i64 + offset) as usize;
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn poll_complete(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<std::io::Result<u64>> {
+        Poll::Ready(Ok(0))
```

Comment:
wrong return value

Reply:
Same as above, would love some input on what it should be. I'm assuming the delta from the initial position which in a simplified version returns the seek offset?

Reply:
Yes, I think this is the delta, so we need to store it and return it. Later when we do seeking with loading, this will also delegate to an internal future.

Reply:
So I took some time to understand this and as far as I get it it's not the delta but rather the final position it seeked to. Because otherwise the consumer would need to figure out what the delta means especially in circular buffer and negative moves etc.

Reply:
Yes new offset from the start is what should be returned 👍